### PR TITLE
Break apart automations doc

### PR DESCRIPTION
**Pull request by kbx81**
## Description:

Breaks the "Automations" guide out into various parts/pages that now have their own section on the front page. This makes them easier to find in general, especially components like `scripts` and `globals`.

Depends on esphome/esphome#6929

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ ] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [x] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Links added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: automations/actions.rst [hunk 1]
```diff
@@ -963,37 +458,164 @@ Configuration variables:
...
+    on_...:
+      then:
+        - if:
+            condition:
+              # Should return either true or false
+              lambda: |-
+                return id(some_sensor).state < 30;
+            # ...
 
+All Triggers
```
Comment (jesserockz) [inline automations/actions.rst]: I think we should drop these blocks, they are missing a lot and not always kept up to date.

File: automations/actions.rst [hunk 2]
```diff
@@ -963,37 +458,164 @@ Configuration variables:
...
+
+.. _automation-networkless:
+
+Do Automations Work Without a Network Connection
+************************************************
+
+This is a common question and the answer is **YES!** All automations you define in ESPHome are executed on the ESP
+itself and will continue to work even if the Wi-Fi network is down or the MQTT server is not reachable.
+
+There is one caveat though: ESPHome automatically reboots if no connection to the MQTT broker can be
```
Comment (jesserockz) [inline automations/actions.rst]: Maybe reword this for API instead of MQTT?

File: automations/interval.rst
Comment (jesserockz) [inline automations/interval.rst:1]: As this is actually a component, maybe it should live as `components/interval.rst` (same with scripts and globals)

Comment (kbx81) [reply automations/interval.rst:1]: Done 😄

File: automations/actions.rst (hunk 1 above)
Comment (kbx81) [reply automations/actions.rst]: There were several references to the `conditions` section -- removed in [`ce18c41` (#3957)](https://github.com/esphome/esphome-docs/pull/3957/commits/ce18c4125a011c76181b482ff0e1b8d787eef7f2) in case you're curious what/where they were.

Comment (jesserockz) [reply automations/actions.rst]: Maybe link to `common_conditions`?

Comment (jesserockz) [reply automations/actions.rst]: Or perhaps `Conditions` (and Triggers) needs its own header (or page) where a general explanation of them plus the common ones can go.

File: components/globals.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        initial_value: '0'
+      # Example for global string variable
+      - id: my_global_string
+        type: std::string
+        restore_value: yes
+        max_restore_data_length: 24
+        initial_value: '"Global value is"'
+
+   # In an automation
+   on_press:
```
Comment (jesserockz) [inline components/globals.rst]:
```suggestion
   on_...:
```

File: components/scripts.rst
Comment (jesserockz) [inline components/scripts.rst:1]: The filename should match the actual component name here
`components/script.rst`

Comment (kbx81) [reply components/scripts.rst:1]: Done in [`aad3714` (#3957)](https://github.com/esphome/esphome-docs/pull/3957/commits/aad3714354368d810d86f68bbf64c52fe49f9289)

**Review by jesserockz — APPROVED** [review]
